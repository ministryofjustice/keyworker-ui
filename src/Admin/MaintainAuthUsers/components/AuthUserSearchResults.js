import React from 'react'
import PropTypes from 'prop-types'
import Button from '@govuk-react/button'
import Table from '@govuk-react/table'
import { GREY_3, BLACK } from 'govuk-colours'
import AuthUserSearch from './AuthUserSearch'
import { authUserListType, errorType } from '../../../types'

const AuthUserSearchResults = props => {
  const { user, userList, handleChange, handleSearch, handleEdit, error } = props
  const results = userList.map((a, index) => (
    <Table.Row key={a.username}>
      <Table.Cell>
        {a.firstName} {a.lastName}
      </Table.Cell>
      <Table.Cell>{a.username}</Table.Cell>
      <Table.Cell>{a.email}</Table.Cell>
      <Table.Cell>{a.locked ? 'Yes' : 'No'}</Table.Cell>
      <Table.Cell>{a.enabled ? 'Yes' : 'No'}</Table.Cell>
      <Table.Cell>
        <Button
          buttonColour={GREY_3}
          buttonTextColour={BLACK}
          mb={0}
          id={`edit-button-${a.username}`}
          value={index}
          onClick={handleEdit}
        >
          Edit
        </Button>
      </Table.Cell>
    </Table.Row>
  ))

  return (
    <div>
      <AuthUserSearch handleSearch={handleSearch} handleChange={handleChange} user={user} error={error} />

      {results.length > 0 && (
        <Table>
          <Table.Row>
            <Table.CellHeader>Name</Table.CellHeader>
            <Table.CellHeader>Username</Table.CellHeader>
            <Table.CellHeader>Email</Table.CellHeader>
            <Table.CellHeader>Locked</Table.CellHeader>
            <Table.CellHeader>Enabled</Table.CellHeader>
            <Table.CellHeader>&nbsp;</Table.CellHeader>
          </Table.Row>
          {results}
        </Table>
      )}
    </div>
  )
}

AuthUserSearchResults.propTypes = {
  user: PropTypes.string,
  userList: authUserListType.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSearch: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
  error: errorType.isRequired,
}

AuthUserSearchResults.defaultProps = {
  user: '',
}

export default AuthUserSearchResults
