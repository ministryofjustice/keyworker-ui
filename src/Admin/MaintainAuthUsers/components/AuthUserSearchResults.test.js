import React from 'react'
import renderer from 'react-test-renderer'
import Table from '@govuk-react/table'
import AuthUserSearchResults from './AuthUserSearchResults'

describe('Auth search results form', () => {
  const stubFunc = () => {}
  describe('no results', () => {
    const wrapper = renderer.create(
      <AuthUserSearchResults handleChange={stubFunc} handleSearch={stubFunc} user="" userList={[]} error={[]} />
    )
    it('should render correctly with no user and no results', () => {
      expect(wrapper.toJSON()).toMatchSnapshot()
    })
    it('should not render the table with no results', () => {
      expect(wrapper.root.findAllByType(Table)).toHaveLength(0)
    })
  })
  describe('some results', () => {
    const userList = [
      {
        username: 'AUTH_ADM',
        email: 'auth_test2@digital.justice.gov.uk',
        firstName: 'Auth',
        lastName: 'Adm',
        locked: true,
        enabled: false,
      },
      {
        username: 'AUTH_EXPIRED',
        email: 'auth_test2@digital.justice.gov.uk',
        firstName: 'Auth',
        lastName: 'Expired',
        locked: false,
        enabled: true,
      },
    ]
    const wrapper = renderer.create(
      <AuthUserSearchResults
        handleChange={stubFunc}
        handleSearch={stubFunc}
        user="someuser"
        userList={userList}
        error=""
      />
    )

    it('should render correctly with user and results', () => {
      expect(wrapper.toJSON()).toMatchSnapshot()
    })
    it('should render the table with a row for each result', () => {
      expect(wrapper.root.findAllByType(Table.Row)).toHaveLength(3)
    })
  })
})
