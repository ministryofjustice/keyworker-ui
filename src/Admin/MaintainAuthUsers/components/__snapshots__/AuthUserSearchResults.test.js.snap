// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth search results form no results should render correctly with no user and no results 1`] = `
<div>
  <div
    className="sc-VigVT dceoHE"
  >
    <div
      className="sc-jTzLTM jTliGH"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="sc-bZQynM wCZqw"
          htmlFor="user"
        >
          <span
            className="sc-gzVnrw clQBNW"
          >
            Username or email address
          </span>
          <input
            className="sc-iwsKbI dkBthv"
            defaultValue=""
            id="user"
            name="user"
            onChange={[Function]}
            type="text"
          />
        </label>
        <button
          className="sc-gZMcBi iEwQUW"
          disabled={false}
          id="search-button"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Auth search results form some results should render correctly with user and results 1`] = `
<div>
  <div
    className="sc-VigVT dceoHE"
  >
    <div
      className="sc-jTzLTM jTliGH"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="sc-bZQynM wCZqw"
          htmlFor="user"
        >
          <span
            className="sc-gzVnrw clQBNW"
          >
            Username or email address
          </span>
          <input
            className="sc-iwsKbI dkBthv"
            defaultValue="someuser"
            id="user"
            name="user"
            onChange={[Function]}
            type="text"
          />
        </label>
        <button
          className="sc-gZMcBi iEwQUW"
          disabled={false}
          id="search-button"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <table
    className="sc-EHOje kCMYgv"
  >
    <tbody
      className="sc-bwzfXH uTxCW"
    >
      <tr
        className="sc-ifAKCX cFlEyZ"
      >
        <th
          className="sc-bxivhb kZbmwT"
        >
          Name
        </th>
        <th
          className="sc-bxivhb kZbmwT"
        >
          Username
        </th>
        <th
          className="sc-bxivhb kZbmwT"
        >
          Email
        </th>
        <th
          className="sc-bxivhb kZbmwT"
        >
          Locked
        </th>
        <th
          className="sc-bxivhb kZbmwT"
        >
          Enabled
        </th>
      </tr>
      <tr
        className="sc-ifAKCX cFlEyZ"
      >
        <td
          className="sc-bxivhb jMmQxS"
        >
          Auth
           
          Adm
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          AUTH_ADM
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          auth_test2@digital.justice.gov.uk
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          Yes
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          No
        </td>
      </tr>
      <tr
        className="sc-ifAKCX cFlEyZ"
      >
        <td
          className="sc-bxivhb jMmQxS"
        >
          Auth
           
          Expired
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          AUTH_EXPIRED
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          auth_test2@digital.justice.gov.uk
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          No
        </td>
        <td
          className="sc-bxivhb jMmQxS"
        >
          Yes
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
